src/wp-includes/class-wp-term.php

```diff
@@ -226,20 +336,28 @@ public function to_array() {
 	 * Getter.
 	 *
 	 * @since 4.4.0
+	 * @since 6.7.0 Getting a dynamic property is deprecated.
 	 *
 	 * @param string $key Property to get.
 	 * @return mixed Property value.
 	 */
-	public function __get( $key ) {
-		switch ( $key ) {
-			case 'data':
-				$data    = new stdClass();
-				$columns = array( 'term_id', 'name', 'slug', 'term_group', 'term_taxonomy_id', 'taxonomy', 'description', 'parent', 'count' );
-				foreach ( $columns as $column ) {
-					$data->{$column} = isset( $this->{$column} ) ? $this->{$column} : null;
-				}
+	public function __get( $name ) {
+		if ( 'data' === $name ) {
+			$data    = new stdClass();
+			$columns = array( 'term_id', 'name', 'slug', 'term_group', 'term_taxonomy_id', 'taxonomy', 'description', 'parent', 'count' );
+			foreach ( $columns as $column ) {
+				$data->{$column} = isset( $this->{$column} ) ? $this->{$column} : null;
```

Comment:
```suggestion
				$data->{$column} = $this->{$column} ?? null;
```

Reply:
Hello @mukeshpanchal27, I don't think that change fits within the scope of this Trac ticket, which is to handle the dynamic properties.

There is a [Trac ticket](https://core.trac.wordpress.org/ticket/58874) open specifically for the change you're proposing. I think it fits better in that ticket's scope.